core/src/SYCL/Kokkos_SYCL_DeepCopy.hpp

```diff
@@ -53,6 +53,28 @@
 namespace Kokkos {
 namespace Impl {
 
+template <class DT, class... DP>
+struct ZeroMemset<Kokkos::Experimental::SYCL, DT, DP...> {
+  static void execute(const Kokkos::Experimental::SYCL& exec_space,
+                      const View<DT, DP...>& dst,
+                      typename View<DT, DP...>::const_value_type&) {
+    auto event = exec_space.impl_internal_space_instance()->m_queue->memset(
+        dst.data(), 0,
+        dst.size() * sizeof(typename View<DT, DP...>::value_type));
+    // FIXME_SYCL this should rather be
+    // exec_space.impl_internal_space_instance()->m_queue->submit_barrier();
+    // but that gives currently a segfault using the CUDA backend
+    Experimental::Impl::SYCLInternal::fence(event);
+  }
+
+  static void execute(const View<DT, DP...>& dst,
+                      typename View<DT, DP...>::const_value_type&) {
+    Experimental::Impl::SYCLInternal::singleton().m_queue->memset(
+        dst.data(), 0,
+        dst.size() * sizeof(typename View<DT, DP...>::value_type));
```

Comment:
I don't think the fencing behavior would be correct if were to defer to the other overload. The three-argument `deep_copy` is supposed to only enqueue the deep copy operation in the same queue whereas the two-argument `deep_copy` fences all execution spaces.
`SYCL` kernels by default don't enqueue at all so that we need some kind of `fence` for the three-argument `deep_copy` but can rely on the general one for the two-argument variant.

In short, deferring would overfence the two-argument `deep_copy`.

Reply:
I overlooked it.  OK